src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -744,7 +748,25 @@ impl Step for LldWrapper {
             extra_features: Vec::new(),
             allow_features: "",
             cargo_args: Vec::new(),
-        })
+        });
+
+        let libdir_bin = builder.sysroot_target_bindir(
+            Compiler { stage: self.compiler.stage + 1, host: self.target },
```

Comment:
Question: this is off-by-one stage, is that intentional? If so, could you leave a comment for why?